src/Track/Processor.php

```diff
@@ -93,23 +106,22 @@ public function process(string $source, Version $version)
                         $point->setElevation(floatval($trackPoint->ele));
                     }
 
+                    $elevationChange = 0;
                     if ($previousPoint) {
                         $distance = $point->distance($previousPoint);
                         $point->setDistance($distance + $previousPoint->getDistance());
 
-                        if ($previousElevation && $point->getElevation()) {
-                            if ($point->getElevation() > $previousElevation) {
-                                $elevationChange = $point->getElevation() - $previousElevation;
+                        if ($previousElevationPoint && $point->getElevation()) {
+                            if ($point->getElevation() > $previousElevationPoint->getElevation()) {
+                                $elevationChange = $point->getElevation() - $previousElevationPoint->getElevation();
                                 if ($previousRealPoint === null || $this->isPointReal($previousRealPoint, $point)) {
-                                    $positiveElevation += $elevationChange;
                                     $previousRealPoint = $point;
                                 } else {
                                     $point->setElevation();
                                 }
                             } else {
-                                $elevationChange = $previousElevation - $point->getElevation();
+                                $elevationChange = $previousElevationPoint->getElevation() - $point->getElevation();
```

Comment:
Is it possible *$previousElevationPoint* to be null here? E.g. *previousPoint* is set set, but *getElevation* returns null?

Reply:
No, previousElevationPoint always have elevation, see https://github.com/trackhub/website/pull/198/files#diff-88877c04d979dfdbd6321a823af1462fR138